src/Microsoft.TemplateEngine.Cli/Commands/BaseCommand.cs

```diff
@@ -92,7 +92,7 @@ protected IEngineEnvironmentSettings CreateEnvironmentSettings(GlobalArgs args,
             };
 
             ParseResult reparseResult = ParserFactory
-                .CreateParser(helperCommand)
+                .CreateParser(helperCommand, disableHelp: true)
```

Comment:
Nit: A comment to explain why we disable help could be useful.

Reply:
we need to pass `-h` through, if `-h` is given it would be intercepted on the first parse, hence disabling the help.